src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -220,20 +252,28 @@ class LazyMobXTableStore<T> {
 
         // add rows (including hidden columns)
         this.dataStore.sortedData.forEach((rowData:T) => {
-            const rowDownloadData:string[] = [];
+            const rowDownloadData:any[] = []; //string[]
+            let refLength = getRefLength(this.columns, rowData);
```

Comment:
okay it's almost there - the one remaining comment I have on this is the refLength approach is inefficient - you're generating the download data twice for each row. You could equivalently just use the length of the first column download - since we're not padding, we're requiring the same length, there's no need to find the max length to compare against.

Reply:
okay I just realized the other reason for finding the max - duplicating non-array download outputs. I think there should be a way to do this without generating all the download data twice.